R/makeTreeSummarizedExperimentFromBiom.R

```diff
@@ -280,30 +267,18 @@ makeTreeSummarizedExperimentFromBiom <- function(obj, ...){
         x,
         patterns,
         invert=FALSE) {
-    if (is(x, "list")) {
-        patterns <- lapply(x, function(x_sub) {
-            grep(patterns,
-                 x_sub[[1]] %>% stringr::str_split("") %>% unlist(),
-                 invert = invert, value = TRUE) %>% unique()
-        }) %>% unlist() %>% unique() %>% paste0(collapse = "")
-    } else if (is(x, "data.frame")){
-        patterns <- apply(x, 2, function(x_sub) {
-            grep(patterns,
-                 x_sub %>% stringr::str_split("") %>% unlist(),
-                 invert = invert, value = TRUE) %>% unique()
-        }) %>% unlist() %>% unique() %>% paste0(collapse = "")
-    }
+    patterns <- apply(x, 2, function(x_sub) {
+        grep(patterns,
+             x_sub %>% stringr::str_split("") %>% unlist(),
+             invert = invert, value = TRUE) %>% unique()
+    }) %>% unlist() %>% unique() %>% paste0(collapse = "")
     return(patterns)
 }
 
 .clean_from_artifacts <- function(x, patterns) {
-    if (is(x, "list")) {
-        x <- lapply(x, gsub, pattern = patterns, replacement = "")
-    } else if (is(x, "data.frame")) {
-        x <- apply(x, 2, gsub, pattern = patterns, replacement = "")
-    }
```

Comment:
Input can only be data.frame

Reply:
True after I change the place where I cast all to data.frame i'll update the utility functions